### PR TITLE
Passing HostConfig for /start endpoint

### LINKED ISSUE #1781: Start container command not forwarding the HostConfig to the container
Client tools (Docker plugin for Jenkins in our case) that rely on passing (additional) HostConfig in the start container command do not work since 1.1.0. For example the port bindings are not passed and the container is inaccessible.

I believe the commit for 'Add support for container rescheduling on node failure.' (https://github.com/docker/swarm/commit/13f60212f57dae8ba480c28bf26261bd7331ea4) broke this flow.

From the Docker Remote API docs:
> 
Start a container:
For backwards compatibility, this endpoint accepts a HostConfig as JSON-encoded request body.
Reference: https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#start-a-container

### PR DESCRIPTION
**Pull request by nishanttotla**
Fixes #1781. `HostConfig` is being deprecated for the `/start` endpoint, but we must support it until Engine does.

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -747,7 +754,7 @@ func postContainersStart(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := c.cluster.StartContainer(container); err != nil {
+	if err := c.cluster.StartContainer(container, &hostConfig); err != nil {
```
Comment (nishanttotla) [inline api/handlers.go]: @vieux how about something like this?

**Comment by nishanttotla** [comment]
Tests are failing because of a single issue - `Error response from daemon: EOF` with error code 125 when running containers. In addition, I see `http: multiple response.WriteHeader calls` in the Swarm console. Pretty sure it's a small issue, so any help debugging is appreciated!

**Comment by nishanttotla** [comment]
Ping @vieux @dongluochen

**Comment by vieux** [comment]
LGTM

**Comment by nishanttotla** [comment]
Test failure seems unrelated, also tests pass locally.

**Comment by dongluochen** [comment]
LGTM.

**Pull request merged**